src/mui/layout/withProps.js

```diff
@@ -0,0 +1,5 @@
+import React from 'react';
+
+export default additionalProps => LayoutComponent => props => (
```

Comment:
If it's so generic, why put it under `layout`, and why not name the component `<Component>` instead?